pytensor/tensor/nlinalg.py

```diff
@@ -200,6 +185,8 @@ class Det(Op):
     """
 
     __props__ = ()
+    gufunc_signature = "(m,m)->()"
```

Comment:
Opening a comment to discuss the signatures, since that was a major bone of contention in the last meeting.

I understood @aseyboldt 's position essentially to be (please correct me if I'm wrong) that the signatures represent only partial information, so they more confusing than helpful in the overall design of an `Op`. For example, you could not use this signature to construct a `numba` function signature from an `Op` alone. Secondly, the signatures don't actually contain the information needed by `Blockwise`. Instead, `Blockwise` has to reconstruct what it needs from the signatures, which are only useful in this one context anyway (see point 1), so their purpose ends up obscured. 

My two cents: I like that it matches the numpy syntax. Perhaps the name could be changed to make it more obvious that these are information specifically to be used by blockwise? Like `blockwise_signature`, or `blockwise_core_sig`? 

There was also a discussion about whether this information should live in the Ops, or whether it should be with Blockwise. I think it conceptually makes sense the way it is -- it's a more OOP design pattern this way, since Blockwise "asks" the Op what how it should interact with it.

Reply:
Noting that @ferrine explored more rich signatures in https://github.com/pymc-devs/pytensor/pull/143

I do think that although they can help they will never be "feature complete" because they would probably require being turing complete :)